app/scripts/controllers/transactions.js

```diff
@@ -60,17 +59,14 @@ module.exports = class TransactionController extends EventEmitter {
     this.pendingTxTracker = new PendingTransactionTracker({
       provider: this.provider,
       nonceTracker: this.nonceTracker,
+      retryLimit: 3500, // Retry 3500 blocks, or about 1 day.
       getBalance: (address) => {
         const account = this.ethStore.getState().accounts[address]
         if (!account) return
         return account.balance
       },
-      publishTransaction: this.query.sendRawTransaction,
+      publishTransaction: (rawTx) => this.query.sendRawTransaction(rawTx),
```

Comment:
why do we do this?
seems like `PendingTransactionTracker` should use its own query around provider to submit the tx